packages/flutter/lib/src/widgets/image.dart

```diff
@@ -464,7 +464,13 @@ class Image extends StatefulWidget {
     this.filterQuality = FilterQuality.low,
     int? cacheWidth,
     int? cacheHeight,
-  }) : image = ResizeImage.resizeIfNeeded(cacheWidth, cacheHeight, FileImage(file, scale: scale)),
+  }) :
+       // FileImage is not supported on Flutter Web therefore neither this method.
+       assert(
+          !kIsWeb,
+          'Image.file is not supported on Flutter Web. '
+          'Consider using either Image.asset or Image.network instead.'),
```

Comment:
Style guide nits:
```suggestion
       assert(
         !kIsWeb,
         'Image.file is not supported on Flutter Web. '
         'Consider using either Image.asset or Image.network instead.',
        ),
```

For more guidance you can read about the Flutter style guide here: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo